src/iuse.cpp

```diff
@@ -2131,6 +2131,52 @@ int iuse::unpack_item( player *p, item *it, bool, const tripoint & )
     return 0;
 }
 
+int iuse::pack_cbm( player *p, item *it, bool, const tripoint & )
+{
+    item_location bionic = g->inv_map_splice( []( const item & e ) {
+        return e.is_bionic();
+    }, _( "Choose CBM to pack" ), PICKUP_RANGE, _( "You don't have any CBMs." ) );
+
+    if( !bionic ) {
+        return 0;
+    }
+    if( !bionic.get_item()->faults.empty() ) {
+        if( p->query_yn( _( "This CBM is faulty.  You should mend it first.  Do you want to try?" ) ) ) {
+            p->mend_item( std::move( bionic ) );
+        }
+        return 0;
+    }
+
+
+    if( !bionic.get_item()->has_flag( "NO_PACKED" ) ||
+        bionic.get_item()->has_flag( "PACKED_FAULTY" ) ) {
+        if( !p->query_yn( _( "This CBM is already prepared.  Do you want to re-do it?" ) ) ) {
+            return 0;
+        }
+    }
+
+    const int success = p->get_skill_level( skill_firstaid ) - rng( 0, 6 );
+    if( success > 0 ) {
+        bionic.get_item()->unset_flag( "NO_PACKED" );
```

Comment:
The condition above on line 2151 is checking for **absence** of `NO_PACKED` flag. Why removing non-existent flag? Or is it a typo in condition and it should check for **existence** of `NO_PACKED` flag?

Here and below on line 2164.

Reply:
Abscence of flag is the default and means the cbm is packed. Salvaged CBM have the flag because they are not packed, when you apply the pouch to them you remove the flag and then they are packed.
Making it backward like this was the easiest way to have sterile and packed CBM by default.

Reply:
Look: we check for absence of flag. It it's true (flag is absent), we're removing the flag (which is, again, absent).

Reply:
The check on 2151 is in case the CBM is already packed and the player want to re pack it again for some reason
One being the packaged was faulty

Reply:
The check on 2151 is in case the CBM is already packed and the player want to re pack it again for some reason
One being the packaged was faulty

Reply:
I don't know how to explain it better. I don't question your intentions, but the logic you applied here is erroneous as you're trying to remove the flag that CBM doesn't have already.

Reply:
Maybe it's that the check for the absence of NO_PACKED flag is unecessary but I do it anyway so that the query appears both for a properly packed CBM and a FAULTY_PACKED cbm so that the player can't know if the package is faulty just by trying to apply the pouch again?

Reply:
Is this better?

Reply:
In this variant the check for flags is redundant. You could just drop the checks (for `NO_PACKED` and `PACKED_FAULTY` flags) and write it as 
```
bionic.get_item()->unset_flag( "NO_PACKED" );
bionic.get_item()->unset_flag( "PACKED_FAULTY" );
```